card/mockCard.go

```diff
@@ -793,6 +944,131 @@ func (c *MockCard) ReceivePhonons(transaction []byte) (err error) {
 	return nil
 }
 
+func (c *MockCard) ReceivePostedPhonons(postedPacket []byte) (err error) {
+	log.Debug("mock RECEIVE_POSTED_PHONONS command")
+
+	collection, err := tlv.ParseTLVPacket(postedPacket)
+	if err != nil {
+		return err
+	}
+
+	nonceBytes, err := collection.FindTag(TagNonce)
+
+	if err != nil {
+		return err
+	}
+
+	nonce := binary.BigEndian.Uint64(nonceBytes)
+
+	// check that nonce is valid
+	if nonce <= c.nonce {
+		return errors.New("transaction.nonce is less than or equal to card.postedPhononNonce")
+	}
+
+	senderCardCertRaw, err := collection.FindTag(TagCardCertificate)
+	if err != nil {
+		return errors.New("could not find certificate tlv tag")
+	}
+
+	senderCardCert, err := cert.ParseRawCardCertificate(senderCardCertRaw)
+	if err != nil {
+		return err
+	}
+
+	// check that sender's cert is valid
+	err = cert.ValidateCardCertificate(senderCardCert, gridplus.SafecardDevCAPubKey)
+	if err != nil {
+		return err
+	}
+
+	senderPubKey, err := util.ParseECCPubKey(senderCardCert.PubKey)
+	if err != nil {
+		return err
+	}
+
+	// check that sender's public key is valid
+	pubKeyValid := gridplus.ValidateECCPubKey(senderPubKey)
+	if !pubKeyValid {
+		return errors.New("counterparty public key is not valid ECC point")
+	}
+
+	phononTransferPacketTLV, err := tlv.ParseTLVPacket(postedPacket, TagTransferPhononPacket)
+	if err != nil {
+		return err
+	}
+
+	phononTLVs, err := phononTransferPacketTLV.FindTags(TagPhononPrivateDescription)
+	if err != nil {
+		return err
+	}
+
+	// parse all received phonons
+	var phonons []MockPhonon
+	for _, phononTLV := range phononTLVs {
+
+		parsedPhononTLV, err := tlv.ParseTLVPacket(phononTLV, TagPhononPrivateDescription)
+		if err != nil {
+			return err
+		}
+
+		encryptedPrivateKey, err := parsedPhononTLV.FindTag(TagPhononPrivKey)
+		if err != nil {
+			log.Debug("could not parse phonon private key tlv")
+			return err
+		}
+
+		iv, err := parsedPhononTLV.FindTag(TagAesIV)
+		if err != nil {
+			log.Debug("could not parse phonon iv tlv")
+			return err
+		}
+
+		privateKey, err := crypto.DecryptData(encryptedPrivateKey, c.identityKey.X.Bytes(), iv)
+		if err != nil {
+			return err
+		}
+
+		phonon, err := decodePostedPhononTLV(phononTLV, privateKey)
+
+		if err != nil {
+			return err
+		}
+		phonons = append(phonons, phonon)
+	}
+
+	sig, err := collection.FindTag(TagECDSASig)
+	if err != nil {
+		return err
+	}
+
+	// here I am converting the data into the format required for the signature verification
+	// there is probably an better way of doing this but I can't work it out.
+	var phononData []byte
+	for _, phononTLV := range phononTLVs {
+		phononTVL, err := tlv.NewTLV(TagPhononPrivateDescription, phononTLV)
+		if err != nil {
+			return err
+		}
+		phononData = append(phononData, phononTVL.Encode()...)
+	}
+
+	// Validate sig
+	isSigValid := ecdsa.VerifyASN1(senderPubKey, CreatePostedPhononSignatureData(c.IdentityPubKey.X.Bytes(), nonceBytes, phononData), sig)
```

Comment:
another spot for ethcrypto.FromECDSAPub()

X.Bytes is not quite correct because there is both an X and Y component to public keys, so this is only half the key. 

Doing it yourself with `append(c.IdentityKey.X.Bytes, c.IdentityKey.Y.Bytes()...)` is almost correct, but I think has some problems in edge cases and doesn't prepend the 0x04 required by ASN1 format.